src/main/java/Duke.java

```diff
@@ -1,10 +1,342 @@
+            } catch (DukeException de) {
+                handleException(line);
+            }
+        }
     }
+
+    /**
+     * Returns the number of tasks found in the list, after handling the commands given by the
+     * user in the form of a line of string.
+     *
+     * @param line     The single line of string inputted by the user.
+     * @param list     The array containing all information about existing tasks and their completion status.
+     * @param listSize The current number of tasks populating the "list" array.
+     * @return Returns the current size of the list populated by tasks created by the user.
+     * @throws DukeException The custom defined exception relating to all errors specific to the Duke program.
+     */
+    private static int handleUserInputs(String line, Task[] list, int listSize) throws DukeException {
+        String[] command = line.split(" ");
+        //Check the first word in the line of strings (list, mark, unmark, event, etc.)
+        switch (command[0]) {
+        case "list":
+            printList(list, listSize, command);
+            break;
+        case "mark":
+            markAsDone(list, listSize, command);
+            break;
+        case "unmark":
+            markAsUndone(list, listSize, command);
+            break;
+        case "todo":
+            if (command.length == 1) {
+                throw new DukeException();
+            }
+            list[listSize] = new ToDo(line.substring(line.indexOf(' ') + 1));
+            listSize = printNewlyAddedTask(list, listSize);
+            break;
+        case "deadline":
+            createNewDeadline(line, list, listSize);
+            listSize = printNewlyAddedTask(list, listSize);
+            break;
+        case "event":
+            createNewEvent(line, list, listSize);
+            listSize = printNewlyAddedTask(list, listSize);
+            break;
+        case "bye":
+            goodbyeMessage(command);
+            break;
+        default:
+            throw new DukeException();
+        }
+        return listSize;
+    }
+
+    /**
+     * Prints the "list" array, which contains a series of user-created tasks. The tasks will be printed
+     * in entry order, starting from the task that was entered into the list first.
+     *
+     * @param list     The array containing all information about existing tasks and their completion status.
+     * @param listSize The current number of tasks populating the "list" array.
+     * @param command  The string array containing all individual strings separated by a space (" ") character in
+     *                 the user inputted string.
+     * @throws DukeException The custom defined exception relating to all errors specific to the Duke program.
+     */
+    private static void printList(Task[] list, int listSize, String[] command) throws DukeException {
+        if (command.length == 1) {
+            lineSeparator();
```

Comment:
`lineSeparator()` seems to be repeated multiple times throughout your code. Maybe try abstracting that into a new function to call.